lib/internal/http2/core.js

```diff
@@ -1077,6 +1077,8 @@ function onHandleFinish() {
   if (this[kID] === undefined) {
     this.once('ready', onHandleFinish.bind(this));
   } else {
+    if (this[kState].headersSent === false)
+      this.rstWithNoError();
```

Comment:
Yea I don't feel great about this either...

Some of the tests send a single whitespace. I thought this was a workaround for this `response.end()` "bug" but maybe it was just a clever way to send some headers and a minimal body.

This was the intended effect:

```js
// Sends HEADERS and DATA(END_STREAM)
require('http2').createServer((request, response) => {
  response.end(' ')
}).listen(8080)

// Sends RST_STREAM(NO_ERROR)
require('http2').createServer((request, response) => {
  response.end() // or: response.end(null)
}).listen(8080)
```

But it could be argued that this is more intuitive:

```js
// Sends HEADERS(END_STREAM)
require('http2').createServer((request, response) => {
  response.end() // or: response.end(null)
}).listen(8080)
```

Sorry to nitpick on such a minor edge-case.